File: fisher-yates-shuffle.rb
def shuffle(arr)

    n = arr.length 

    (0..n-1).each do |i|

        r = Random.rand(i+1)

        arr[i], arr[r] = arr[r],arr[i]  

    end 
end

l = [1,2,3,4,5,6,7,8,9]

print l

shuffle(l)

puts ""

print l

File: fisher-yates-shuffle.php
<?php
/**
 *  It's an algorithm for generating a random permutation of a finit sequence - in plain terms, the algorithm shuffles the sequence.
 *
 *  Reference: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
 */


function MyShuffle(&$arr) {

         for($i = 0; $i < sizeof($arr); ++$i) {
             $r = rand(0, $i);
             $tmp = $arr[$i];
             $arr[$i] = $arr[$r];
             $arr[$r] = $tmp;  
         }
};

echo"<h2>Hello Fisher-Yates Shuffle Algorithm</h2>";

$arr = array(0,1,2,3,4,5,6,7,8,9);

echo"<pre>";
print_r($arr);
echo"</pre>";

MyShuffle($arr);

echo"<pre>";
print_r($arr);
echo"</pre>";
      
?>